pkg/storegateway/bucket_streaming_readers.go

```diff
@@ -133,6 +138,8 @@ func (s *SeriesChunksStreamReader) StartBuffering() {
 				// which is true at the time of writing.
 				s.errorChan <- s.client.Context().Err()
 				return
+			case <-s.closeCalled:
+				return
```

Comment:
Can't we rely on the context being closed here?

Reply:
This is for when we want to close the stream readers ourselves. At the point where we need to close it, there is no cancel functions for the context. If we use `context.WithCancel`, IDE gives a warning that there might be context leak because we won't cancel the context in all cases.

We want to close the stream here https://github.com/grafana/mimir/blob/1720db083725d6128c972587c74340c137c19f1f/pkg/querier/blocks_store_queryable.go#L472-L476

Reply:
OK, makes sense, thanks.